math/factorial.cpp

```diff
@@ -1,20 +1,59 @@
 /**
  * @file
- * @brief C++ program to find factorial of given number
+ * @brief Find the [factorial](https://en.wikipedia.org/wiki/Factorial) of a given number
+ * @description calculates factorial via recursion
+ * \f[n! = n\times(n-1)\times(n-2)\times(n-3)\times\ldots\times3\times2\times1
+ *       = n\times(n-1)!\f]
+ * for example:
+ * \f$5! = 5\times4! = 5\times4\times3\times2\times1 = 120\f$
+ *
+ * @author [Akshay Gupta](https://github.com/Akshay1910)
  */
-#include <iostream>
 
-/** function to find factorial of given number */
-unsigned int factorial(unsigned int n) {
-    if (n == 0)
+#include <cstdint>   /// for uint64_t
```

Comment:
AFAIK, `uint_*` types are already included without the need of any external libraries.

Reply:
ah makes sense my ide keeps freaking out when i dont include it